src/hotspot/share/gc/g1/heapRegion.cpp

```diff
@@ -90,6 +90,9 @@ void HeapRegion::setup_heap_region_size(size_t max_heap_size) {
   guarantee(GrainWords == 0, "we should only set it once");
   GrainWords = GrainBytes >> LogHeapWordSize;
 
+  // Initialize card size
+  CardTable::initialize_card_size();
```

Comment:
I see that `CardTable::initialize_card_size()` is called at the beginning of `GCArguments::initialize_alignments()` for Serial and Parallel. I wonder if it makes sense to do the same for G1.

Reply:
The reason for having `CardTable::initialize_card_size()` here in the initial patch was the dependency between card size and G1 heap region size. Since the dependency is no longer there, we can safely move this initialization to `G1Arguments::initialize_alignments`